platforms/substrate/charts/vitalam-ipfs-node/templates/statefulset.yaml

```diff
@@ -0,0 +1,116 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "vitalam-ipfs.fullname" . }}
+  labels:
+    {{- include "vitalam-ipfs.labels" . | nindent 4 }}
+spec:
+  serviceName: {{ include "vitalam-ipfs.fullname" . }}
+  selector:
+    matchLabels:
+      name: {{ include "vitalam-ipfs.fullname" . }}
+  template:
+    metadata:
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
+        checksum/secret: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
+      labels:
+        name: {{ include "vitalam-ipfs.fullname" . }}
+    spec:
+      {{- include "vitalam-ipfs.imagePullSecrets" . | indent 6 }}
+      initContainers:
+        - name: {{ include "vitalam-ipfs.initIpfsConfig.name" . }}
+          image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
+          imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
+          env:
+            - name: IPFS_PATH
+              value: {{ .Values.config.ipfsDataPath }}
+          envFrom:
+            - configMapRef:
+                name: {{ include "vitalam-ipfs.fullname" . }}-config
+          command: ['/bin/sh', '-c', '-e', '-x']
+          args:
+            - |
+              [ ! -f "$IPFS_PATH/config" ] && $ipfsCommand init --profile server;
+              $ipfsCommand config Addresses.API /ip4/0.0.0.0/tcp/$ipfsApiPort;
+              $ipfsCommand config --json Addresses.Swarm "[\"/ip4/0.0.0.0/tcp/$ipfsSwarmPort\"]";
+              $ipfsCommand config Datastore.StorageMax $(echo "$dataVolumeSize""Gb");
+              $ipfsCommand config --json API.HTTPHeaders.Access-Control-Allow-Origin "[\"http://0.0.0.0:$ipfsApiPort\", \"http://127.0.0.1:$ipfsApiPort\"]";
+              $ipfsCommand bootstrap rm --all;
+              if [ ! -z "$ipfsBootNodeAddress" ]; then $ipfsCommand bootstrap add $ipfsBootNodeAddress; fi;
```

Comment:
These may need to be changed for our deployment as this is used localhost as IPs

Reply:
@sownak what would the values be for our deployment?

Reply:
These can remain for now, you will need to update as part of your next task.